### DISCUSSION TITLE
Edge with no default empty constructor (Edge{})

### DISCUSSION THREAD

**Discussion by IkarusDeveloper** [artem-ogre/CDT#85] [Q&A] [answered]
I m really confused about the fact that Edge seems to doesn't allow any empty initialization, a requirement which any value_type used in a `std::vector` must allow.

But the function (member of Triangulation) is accepting `std::vector<Edge>` as argument.
`void insertEdges(const std::vector<Edge>& edges);
`
Here the interface of `class Edge`.
```
struct CDT_EXPORT Edge
{
    /// Constructor
    Edge(VertInd iV1, VertInd iV2);
    /// Assignment operator
    bool operator==(const Edge& other) const;
    /// V1 getter
    VertInd v1() const;
    /// V2 getter
    VertInd v2() const;
    /// Edges' vertices
    const std::pair<VertInd, VertInd>& verts() const;

private:
    std::pair<VertInd, VertInd> m_vertices;
};
```

What i m not getting here?
Thank you in advance.

Once i added an empty constructor to the class all the compilation errors just disappeared, so i m still confused.
`Edge() = default;`

**Comment by artem-ogre** [comment, marked as answer]
Hi, @IkarusDeveloper,

From `std::vector` [docs](https://en.cppreference.com/w/cpp/container/vector):

> T must meet the requirements of [CopyAssignable](https://en.cppreference.com/w/cpp/named_req/CopyAssignable) and [CopyConstructible](https://en.cppreference.com/w/cpp/named_req/CopyConstructible).

which `Edge` satisfies.

I am not sure what compilation errors you are referring  to.

**Reply by IkarusDeveloper** [reply]
Sorry for bothering you, i've already starred your repo, i really appreciate your effort.
Here's a minimal reproducible example.
https://godbolt.org/z/K1hT9dKbz
It looks `resize` wont work on `Edge`s vectors.
I think you should make it usable, it may optimize some stuff.
And it again happen while inserting with emplace_back using overload with no arguments.
`auto& edge = vector_break.emplace_back();`

**Reply by artem-ogre** [reply]
Yes `resize` would not work as the value needs to be default constructible. It is still possible to use `reserve`, `push_back` and `emplace_back`.

**Comment by IkarusDeveloper** [comment]
So i have to correct myself.
The empty constructor isn't `required` to instantiate a vector of `Edge`.
It it just `required` in order to use some `vector`'s stuffs (vector constructor with size,  `vector.resize`. some overload of `vector.emplace_back` and probably something I'm not thinking about right now)
